lib/reek/cli/options.rb

```diff
@@ -14,110 +9,122 @@ module Cli
     # Parses the command line
     #
     class Options
-      include CLI::Input
-
-      attr_reader :config_file, :smells_to_detect
-
       def initialize(argv)
-        @argv                = argv
-        @parser              = OptionParser.new
-        @colored             = true
-        @report_class        = Report::TextReport
-        @strategy            = Report::Strategy::Quiet
-        @warning_formatter   = Report::WarningFormatterWithLineNumbers
-        @command_class       = ReekCommand
-        @config_file         = nil
-        @sort_by_issue_count = false
-        @smells_to_detect    = []
-        set_options
+        @argv    = argv
+        @parser  = OptionParser.new
+        @options = OpenStruct.new(colored: true, smells_to_detect: [])
+        set_up_parser
+      end
+
+      def parse
+        @parser.parse!(@argv)
+        @options.argv = @argv
+        Rainbow.enabled = @options.colored
+        @options
       end
 
+      private
+
       def banner
-        progname = @parser.program_name
+        program_name = @parser.program_name
         <<-EOB.gsub(/^[ ]+/, '')
-          Usage: #{progname} [options] [files]
+          Usage: #{program_name} [options] [files]
 
           Examples:
 
-          #{progname} lib/*.rb
-          #{progname} -q lib
-          cat my_class.rb | #{progname}
+          #{program_name} lib/*.rb
+          #{program_name} -s lib
+          cat my_class.rb | #{program_name}
 
           See http://wiki.github.com/troessner/reek for detailed help.
 
         EOB
       end
 
-      def set_options
+      def set_up_parser
         @parser.banner = banner
-        @parser.separator 'Common options:'
-        @parser.on('-h', '--help', 'Show this message') do
-          @command_class = HelpCommand
+        set_configuration_options
+        set_alternative_formatter_options
+        set_report_formatting_options
+        set_utility_options
+      end
+
+      def set_utility_options
```

Comment:
Can we order the methods below the way they are used in set_up_parser? Just makes it a little easier to follow the flow.